### PR TITLE
Adding cuda-nvrtc recipe

### PR DESCRIPTION
**Pull request by adibbley**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/cuda-nvrtc/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    - {{ compiler("cxx") }}
+  host:
+    - sysroot_{{ target_platform }} 2.17   # [linux]
+    - __glibc >=2.17                       # [linux]
+  run:
+    - sysroot_{{ target_platform }} 2.17   # [linux]
+    - __glibc >=2.17                       # [linux]
+
+outputs:
+  - name: cuda-nvrtc
```
Comment (bdice) [inline recipes/cuda-nvrtc/meta.yaml:40]:
```suggestion
  - name: cuda-nvrtc
    requirements:
      run:
        - cuda-version =12.0
```

File: recipes/cuda-nvrtc/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      - Library\bin               # [win]
+
+  - name: cuda-nvrtc-dev
+    test:
+      commands:
+        - test -d $PREFIX/lib/pkgconfig                 # [linux]
+        - if not exist %LIBRARY_LIB%\nvrtc.lib exit 1   # [win]
+    requirements:
+      run:
+        - cuda-nvrtc >={{ version }}
```
Comment (bdice) [inline recipes/cuda-nvrtc/meta.yaml:57]:
```suggestion
        - cuda-nvrtc >={{ version }}
        - cuda-version =12.0
```

File: recipes/cuda-nvrtc/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      - Library\lib\nvrtc.lib  # [win]
+
+  - name: cuda-nvrtc-static
+    test:
+      commands:
+        - test -f $PREFIX/lib/libnvrtc_static.a               # [linux]
+        - if not exist %LIBRARY_LIB%\nvrtc_static.lib exit 1  # [win]
+    requirements:
+      run:
+        - cuda-nvrtc-dev >={{ version }}
```
Comment (bdice) [inline recipes/cuda-nvrtc/meta.yaml:75]:
```suggestion
        - cuda-nvrtc-dev >={{ version }}
        - cuda-version =12.0
```

**Review by bdice — COMMENTED** [review]
Added `cuda-version` run dependencies.

File: recipes/cuda-nvrtc/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,99 @@
+{% set name = "cuda-nvrtc" %}
+{% set version = "12.0.76" %}
```
Comment (bdice) [inline recipes/cuda-nvrtc/meta.yaml:2]:
```suggestion
{% set version = "12.0.76" %}
{% set cuda_version = "12.0" %}
```

File: recipes/cuda-nvrtc/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    - __glibc >=2.17                       # [linux]
+  run:
+    - sysroot_{{ target_platform }} 2.17   # [linux]
+    - __glibc >=2.17                       # [linux]
+
+outputs:
+  - name: cuda-nvrtc
+    requirements:
+      run:
+        - cuda-version =12.0
```
Comment (bdice) [inline recipes/cuda-nvrtc/meta.yaml:43]:
```suggestion
        - cuda-version {{ cuda_version }}
```

File: recipes/cuda-nvrtc/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+  - name: cuda-nvrtc-dev
+    test:
+      commands:
+        - test -d $PREFIX/lib/pkgconfig                 # [linux]
+        - if not exist %LIBRARY_LIB%\nvrtc.lib exit 1   # [win]
+    requirements:
+      run:
+        - cuda-nvrtc >={{ version }}
+        - cuda-version =12.0
```
Comment (bdice) [inline recipes/cuda-nvrtc/meta.yaml:61]:
```suggestion
        - cuda-version {{ cuda_version }}
```

File: recipes/cuda-nvrtc/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+  - name: cuda-nvrtc-static
+    test:
+      commands:
+        - test -f $PREFIX/lib/libnvrtc_static.a               # [linux]
+        - if not exist %LIBRARY_LIB%\nvrtc_static.lib exit 1  # [win]
+    requirements:
+      run:
+        - cuda-nvrtc-dev >={{ version }}
+        - cuda-version =12.0
```
Comment (bdice) [inline recipes/cuda-nvrtc/meta.yaml:80]:
```suggestion
        - cuda-version {{ cuda_version }}
```

**Review by robertmaynard — APPROVED** [review]

File: recipes/cuda-nvrtc/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+      build:
+        - {{ compiler("c") }}
+        - {{ compiler("cxx") }}
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+        - sysroot_{{ target_platform }} 2.17    # [linux]
+      host:
+         - cuda-version {{ cuda_version }}
+      run:
+         - {{ pin_compatible("cuda-version", max_pin="x.x") }}
+    test:
```
Comment (bdice) [inline recipes/cuda-nvrtc/meta.yaml:46]: Please merge this PR to fix tests: https://github.com/adibbley/staged-recipes/pull/5

**Review by kkraus14 — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks all! 🙏

Let's follow up on anything else in the feedstock 🙂

**Pull request merged**